pkg/component/component.go

```diff
@@ -373,33 +372,32 @@ func PushLocal(client *occlient.Client, componentName string, applicationName st
 	targetPath := fmt.Sprintf("%s/src", s2iSrcPath)
 
 	// Copy the files to the pod
-
-	s := log.Spinner("Copying files to pod")
+	spin := log.Spinner(true, "Copying files to pod")
 	err = client.CopyFile(path, pod.Name, targetPath, files)
 	if err != nil {
-		s.End(false)
+		spin.End(false)
 		return errors.Wrap(err, "unable push files to pod")
 	}
-	s.End(true)
+	spin.End(true)
 
-	s = log.Spinner("Building component")
+	// Create a spinner for the component being built
+	var s *log.Status
+	if show {
```

Comment:
why not use `show` in `log.Spinnerf` directly?
`s := log.Spinnerf(!show, "Building component")`